### PR TITLE
Fix: grammar.g : constrain main function naming

### LINKED ISSUE #13: Fix: grammar and lexer
# Bug Report
Make grammar and lexer consistent with problem statement.

### Some features are still missing in main branch *grammar* and *lexer*.
Data Type : integer (int), floating point (float) and void
Declaration statements : identifiers are declared in declaration statements as basic data types and may also be assigned constant values (integer of floating)
Condition constructs: if, else, nested statements are supported. There may be if statement without else statement.
Assignments to the variables are performed using the input / output constructs:
        cin >> x - Read into variable x
        cout << x - Write variable x to output 
Only arithmetic operators {+, -, *, %} and assignment operator `=’ are     supported
Relational operators used in the if statement are < (less than), > (greater than), == (equal) and != (not equal)
Only function is main(), there is no other function. The main() function does not contain arguments and no return statements.

### Expected Behavior
Rule for following operators should be present in grammar:
- %

Support for additional tokens: 
 - \>=
 - <=
 - !=
 - \+
 - \-
 - \*
 - /
 - %

## Solution
Update grammar.g file and lexer.

### PR DESCRIPTION
**Pull request by noob77777**
# Updating grammar and lexer

## Description
Adds improvements and fixes to lexer and parsing grammar.
resolves #13 

## Remaining tasks

- [x] Fix grammar
- [x] Update lexer

### CODE REVIEW

**Comment by noob77777** [comment]
commit history fuck up korchish kno

File: test/lexer/lexer_test.cpp [hunk 1]
```diff
@@ -207,8 +207,6 @@ TEST(lexer, lexer4) {
   ASSERT_EQ(jucc::lexer::TOK_CURLY_CLOSE, token);
   token = lex.GetToken(is);
   ASSERT_EQ(jucc::lexer::TOK_COMMENT, token);
-  token = lex.GetToken(is);
-  ASSERT_EQ(jucc::lexer::TOK_EOF, token);
```
Comment (noob77777) [inline test/lexer/lexer_test.cpp]: 💢

**Comment by TheSYNcoder** [comment]
L

> commit history fuck up korchish kno

Already started working on symbol_table branch, decided to merge it later. Was not a good decision. 🥲

File: test/lexer/lexer_test.cpp (hunk 1 above)
Comment (TheSYNcoder) [reply test/lexer/lexer_test.cpp]: Unintentional ..

Comment (TheSYNcoder) [reply test/lexer/lexer_test.cpp]: Pull and change please.

**Comment by TheSYNcoder** [comment]
Nice it got solved!

**Review by TheSYNcoder — APPROVED** [review]

**Pull request merged**